### ISSUE TITLE
Downloading individual files from torrent

### ISSUE THREAD

**Issue by Suncatcher** [aria2/aria2#843]
It is not completely clear from #658 ticket, if Aria2 at all able to download torrents partially (selected files) or only in full?

**Comment by tatsuhiro-t** [comment]
aria2 can download selected files (see http://aria2.github.io/manual/en/html/aria2c.html#cmdoption--select-file).
aria2 has limited prioritization capability (see http://aria2.github.io/manual/en/html/aria2c.html#cmdoption--bt-prioritize-piece), but it is not based on file.
Does it make sense?

**Comment by Suncatcher** [comment]
Thanks, now it's clear.

**Comment by TrainerRoshi** [comment]
It's not clear to me. It the docs it say: `for example: 3,6` if I want to download files with index 3 and with 6. Where do I add the 3,6 in the command line?

**Comment by AriaMoradi** [comment]
> It's not clear to me. It the docs it say: `for example: 3,6` if I want to download files with index 3 and with 6. Where do I add the 3,6 in the command line?

you use it like this:
`
aria2c --select-file=3,6 your-torrent-file.torrent
`
and you can see the file indexes by using `--show-files`:
`
aria2c --show-files your-torrent-file.torrent
`

**Comment by AriaMoradi** [comment]
I think this issue should be closed.

**Comment by smhmd** [comment]
@AriaMoradi Do you know how to `--show-files` with a magnet link instead of a torrent file?

**Comment by AriaMoradi** [comment]
@smhmd 
no, I don't think aria2 has that feature. but none the less there are free websites that do the conversion for you.
here's one: http://www.magnet2torrent.com/

**Comment by smhmd** [comment]
@AriaMoradi I'm aware of those but it would require leaving the comfort of your terminal. I'd be happy to find a `curl` equivalent or for aria2 to support this (https://github.com/aria2/aria2/issues/1673).

**Comment by anadius** [comment]
@smhmd `aria2c --bt-metadata-only=true --bt-save-metadata=true <magnet link>` should save the torrent file. Then you can use `--show-files`.

**Comment by zamanuhina** [comment]
How to show files using RPC? Or how to show files in JSON?

aria2c --show-files `file.torrent`

**Comment by TVMD** [comment]
> aria2 can download selected files (see http://aria2.github.io/manual/en/html/aria2c.html#cmdoption--select-file). aria2 has limited prioritization capability (see http://aria2.github.io/manual/en/html/aria2c.html#cmdoption--bt-prioritize-piece), but it is not based on file. Does it make sense?

I tried option --select-file but it weir, When I use --select-file=15 . It download 14 and 16 also. Can you explain how can I download only selected file? 
P/s: I tried --select-file=15-15, select-file=15,15. and --select-file=15,. Still the same

**Comment by TrainerRoshi** [comment]
Well no. 14 and 16 aren't downloaded. Try to open them, you'll see it doesn't work. 14 and 16 show up because 15 needed a block it shares with those other two files. Like this:
⬜⬜⬜⬜🔳⬛⬛⬛🔳⬜⬜⬜⬜
